packages/matter.js/src/cluster/GeneralDiagnosticCluster.ts

```diff
@@ -0,0 +1,209 @@
+    VisualDisplayFault = 0x06,
+    AudioOutputFault = 0x07,
+    UserInterfaceFault = 0x08,
+    NonVolatileMemoryError = 0x09,
+    TamperDetected = 0x0A,
+}
+/** @see {@link MatterCoreSpecificationV1_0} § 11.11.6.2 */
+export const enum RadioFault {
+    Unspecified = 0x00,
+    WiFiFault = 0x01,
+    CellularFault = 0x02,
+    ThreadFault = 0x03,
+    NFCFault = 0x04,
+    BLEFault = 0x05,
+    EthernetFault = 0x06,
+}
+/** @see {@link MatterCoreSpecificationV1_0} § 11.11.6.3 */
+export const enum NetworkFault {
+    Unspecified = 0x00,
+    HardwareFailure = 0x01,
+    NetworkJammed = 0x02,
+    ConnectionFaile = 0x03,
+}
+/** @see {@link MatterCoreSpecificationV1_0} § 11.11.6.4 */
+export const enum InterfaceType {
+    Unspecified = 0x00,
+    WiFi = 0x01,
+    Ethernet = 0x02,
+    Celluar = 0x03,
+    Thread = 0x04,
+}
+/** @see {@link MatterCoreSpecificationV1_0} § 11.11.7.5 */
+export const enum BootReason {
+    Unspecified = 0x00,
+    PowerOnReboot = 0x01,
+    BrownOutReset = 0x02,
+    SoftwareWatchdogReset = 0x03,
+    HardwareWatchdogReset = 0x04,
+    SoftwareUpdateCompleted = 0x05,
+    SoftwareReset = 0x06,
+}
+
+/**  describes a network interface supported by the Node, as provided in the NetworkInterfaces attribute. 
+    @see {@link MatterCoreSpecificationV1_0} § 11.11.6.5
+*/
+const TlvNetworkInterface = TlvObject({
+    /** human-readable (displayable) name for the network interface */
+    name: TlvField(0, TlvString.bound({ maxLength: 32 })),
+
+    /** indicate if the Node is currently advertising itself operationally on this network interface */
+    isOperational: TlvField(1, TlvBoolean),
+
+    /** indicate whether the Node is currently able to reach off-premise services it uses by utilizing IPv4 */
+    offPremiseServicesReachableIPv4: TlvField(2, TlvNullable(TlvBoolean)), /* default null */
+
+    /** indicate whether the Node is currently able to reach off-premise services it uses by utilizing IPv4 */
+    offPremiseServicesReachableIPv6: TlvField(3, TlvNullable(TlvBoolean)),  /* default null */
+
+    /** SHALL contain the current link-layer address for a 802.3 or IEEE 802.11-2020 network interface and 
+     * contain the current extended MAC address for a 802.15.4 interface 
+    */
+    hardwareAddress: TlvField(4, TlvString.bound({ maxLength: 8 })),
+
+    /** list of the IPv4 addresses that are currently assigned to the network interface */
+    iPv4Addresses: TlvField(5, TlvArray(TlvString.bound({ maxLength: 4 }))),
+
+    /** list of the unicast IPv6 addresses that are currently assigned to the network interface. */
+    iPv6Addresse: TlvField(5, TlvArray(TlvString.bound({ maxLength: 8 }))),
+
+    /** indicate the type of the interface */
+    type: TlvField(6, TlvEnum<InterfaceType>()),
+});
+
+/** SHALL indicate a change in the set of hardware faults currently detected by the Node
+   * @see {@link MatterCoreSpecificationV1_0} §11.11.9.1
+ */
+const HardwareFaultChangeEvent = {
```

Comment:
These also need to be TlvObject and we should also add Tlv o their name start ...